c++/src/kj/compat/http.c++

```diff
@@ -729,8 +729,17 @@ void HttpHeaders::takeOwnership(HttpHeaders&& otherHeaders) {
 
 // -----------------------------------------------------------------------------
 
-static inline char* skipSpace(char* p) {
+static inline char* skipSpace(char* p, kj::Maybe<size_t> end = nullptr) {
```

Comment:
I think reusing this function may be more trouble than it's worth at this point, if you have to make modifications like this.

Instead you can implement `stripLeadingSpace()` like:

```
void stripLeadingSpace(ArrayPtr<const char>& str) {
  while (str.size() > 0 && (str[0] == ' ' || str[0] == '\t')) {
    str = str.slice(1, str.size());
  }
}
```

Overall keeping it separate ends up being less code anyway.